### PR TITLE
csclient: add the ability to set custom headers.

### PR DESCRIPTION
**Pull request by frankban**
Also update to use the latest httpbakery.

### CODE REVIEW

File: csclient/csclient.go [hunk 1]
```diff
@@ -424,6 +430,11 @@ func (c *Client) DoWithBody(req *http.Request, path string, getBody httpbakery.B
 	if !strings.HasPrefix(path, "/") {
 		return nil, errgo.Newf("path %q is not absolute", path)
 	}
+	for k, vv := range c.header {
+		for _, v := range vv {
```
Comment (mhilton) [inline csclient/csclient.go]: Why not `req.Header[k] = append(req.Header[k], v...)`?

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/891/
Test PASSed.

**Comment by mhilton** [comment]
:+1: LGTM with a suggestion

File: csclient/csclient.go [hunk 2]
```diff
@@ -99,6 +100,11 @@ func (c *Client) DisableStats() {
 	c.statsDisabled = true
 }
 
+// SetHeader sets custom header to be sent to the charm store on each request.
+func (c *Client) SetHeader(header http.Header) {
```
Comment (rogpeppe) [inline csclient/csclient.go]: SetHTTPHeader ?
(the charmstore client isn't really about HTTP, so it's perhaps better to be more specific here)

File: csclient/csclient.go [hunk 3]
```diff
@@ -99,6 +100,11 @@ func (c *Client) DisableStats() {
 	c.statsDisabled = true
 }
 
+// SetHeader sets custom header to be sent to the charm store on each request.
```
Comment (rogpeppe) [inline csclient/csclient.go]: ... sets custom HTTP headers that will be sent to the charm store
on each request.

?

File: csclient/csclient.go (hunk 1 above)
Comment (rogpeppe) [inline csclient/csclient.go]: +1

**Comment by rogpeppe** [comment]
LGTM with a couple of trivials.

File: csclient/csclient.go (hunk 1 above)
Comment (frankban) [inline csclient/csclient.go]: Done.

File: csclient/csclient.go (hunk 2 above)
Comment (frankban) [inline csclient/csclient.go]: Done.

File: csclient/csclient.go (hunk 3 above)
Comment (frankban) [inline csclient/csclient.go]: Done.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/892/
Test PASSed.

**Pull request merged**